src/Staking.sol

```diff
@@ -89,84 +91,99 @@ contract StakeRegistry is AccessControl, Pausable {
      * @notice Create a new stake or update an existing one.
      * @dev At least `_initialBalancePerChunk*2^depth` number of tokens need to be preapproved for this contract.
      * @param _owner Eth address used for overlay calculation.
-     * @param nonce Nonce that was used for overlay calculation.
-     * @param amount Deposited amount of ERC20 tokens.
+     * @param _nonce Nonce that was used for overlay calculation.
+     * @param _amount Deposited amount of ERC20 tokens.
      */
-    function depositStake(address _owner, bytes32 nonce, uint256 amount) external whenNotPaused {
+    function depositStake(address _owner, bytes32 _nonce, uint256 _amount) external whenNotPaused {
```

Comment:
if this allows new nonce (change overlay), why do we need the `changeOverlay` function?

Reply:
yeah, good call. changing this